### ISSUE TITLE
System.MissingMethodException: Method not found: 'System.Text.RegularExpressions.Match System.Text.RegularExpressions.Regex.Match(System.String, System.String)'

### ISSUE THREAD

**Issue by JourneyOver** [Citrinate/FreePackages#34]
So pretty much getting the below error every 15 minutes it looks like going by ASF logs date modified info.. Seems to be something to do with a regex match or something in this plugin..

```log
2024-05-07 06:47:22|ArchiSteamFarm-18|INFO|ASF|InitCore() ArchiSteamFarm V6.0.3.0 (linux-x64/9a807186076740deb0212451941ca9ef | .NET 8.0.4; linux-x64; Debian GNU/Linux 12 (bookworm)) in [/app]
2024-05-07 06:47:22|ArchiSteamFarm-18|INFO|ASF|InitCore() Copyright © 2015-2024 JustArchiNET
2024-05-07 06:47:22|ArchiSteamFarm-18|WARN|ASF|InitGlobalConfigAndLanguage() config/ASF.json config file will be migrated to the latest syntax...
2024-05-07 06:47:22|ArchiSteamFarm-18|INFO|ASF|InitGlobalConfigAndLanguage() Done!
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() Initializing Plugins...
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() Loading ItemsMatcherPlugin V6.0.3.0...
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() ItemsMatcherPlugin has been loaded successfully!
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() Loading SteamTokenDumperPlugin V6.0.3.0...
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() SteamTokenDumperPlugin has been loaded successfully!
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() Loading MobileAuthenticatorPlugin V6.0.3.0...
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() MobileAuthenticatorPlugin has been loaded successfully!
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() Loading AutoClaimStickers V1.0.2.1...
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() AutoClaimStickers has been loaded successfully!
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() Loading ASFEnhance V2.1.4.0...
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() ASFEnhance has been loaded successfully!
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() Loading ASFFreeGamesPlugin V1.5.1.0...
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|OnLoaded() Loaded ASFFreeGamesPlugin
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() ASFFreeGamesPlugin has been loaded successfully!
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() Loading FreePackages V1.4.5.0...
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|OnLoaded() Free Packages ASF Plugin by Citrinate
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() FreePackages has been loaded successfully!
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() You've loaded one or multiple custom plugins into ASF. Since we're unable to offer support for modded setups, please contact the appropriate developers of the plugins that you decided to use in case of any issues.
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() AutoClaimStickers (AutoClaimStickers) plugin has been disabled from automatic updates, despite supporting such feature.
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() ASFEnhance (ASFEnhance) plugin has been disabled from automatic updates, despite supporting such feature.
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|InitPlugins() FreePackages (FreePackages) plugin has been disabled from automatic updates, despite supporting such feature.
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|UpdateAndRestart() ASF will automatically check for new versions every 12 hours.
2024-05-07 06:47:23|ArchiSteamFarm-18|INFO|ASF|UpdateASF() Checking for new version...
2024-05-07 06:47:24|ArchiSteamFarm-18|INFO|ASF|UpdateASF() Local version: 6.0.3.0 | Remote version: 6.0.3.0
2024-05-07 06:47:24|ArchiSteamFarm-18|INFO|ASF|OnASFInit() 
==========================================================================
 █████╗ ███████╗███████╗███████╗███╗   ██╗██╗  ██╗ █████╗ ███╗   ██╗ ██████╗███████╗
██╔══██╗██╔════╝██╔════╝██╔════╝████╗  ██║██║  ██║██╔══██╗████╗  ██║██╔════╝██╔════╝
███████║███████╗█████╗  █████╗  ██╔██╗ ██║███████║███████║██╔██╗ ██║██║     █████╗  
██╔══██║╚════██║██╔══╝  ██╔══╝  ██║╚██╗██║██╔══██║██╔══██║██║╚██╗██║██║     ██╔══╝  
██║  ██║███████║██║     ███████╗██║ ╚████║██║  ██║██║  ██║██║ ╚████║╚██████╗███████╗
╚═╝  ╚═╝╚══════╝╚═╝     ╚══════╝╚═╝  ╚═══╝╚═╝  ╚═╝╚═╝  ╚═╝╚═╝  ╚═══╝ ╚═════╝╚══════╝
==========================================================================
ASFEnhance Version 2.1.4.0
Developed by Chr_, email chr@chrxw.com
Donate: https://afdian.net/@chr233
==========================================================================
Submodule not loaded
==========================================================================
Check for plugins to update using "PLUGINVERSION [插件名]/ "PV [插件名]
Automatically update the specified plugin with "PLUINUPDATE [插件名]" / "PUINUPDATE [插件名]
不指定插件名时, 可为所有支持的插件检查更新 / 安装更新
==========================================================================

2024-05-07 06:47:24|ArchiSteamFarm-18|INFO|ASF|Load() Loading STD global cache...
2024-05-07 06:47:25|ArchiSteamFarm-18|INFO|ASF|Load() Validating STD global cache integrity...
2024-05-07 06:47:25|ArchiSteamFarm-18|INFO|ASF|OnASFInit() SteamTokenDumperPlugin has been initialized successfully, thank you in advance for your help. The first submission will happen in approximately 13 minutes from now.
2024-05-07 06:47:25|ArchiSteamFarm-18|INFO|ASF|Start() Starting IPC server...
2024-05-07 06:47:26|ArchiSteamFarm-18|INFO|Microsoft.Hosting.Lifetime|Now listening on: http://[::]:1242
2024-05-07 06:47:26|ArchiSteamFarm-18|INFO|Microsoft.Hosting.Lifetime|Application started. Press Ctrl+C to shut down.
2024-05-07 06:47:26|ArchiSteamFarm-18|INFO|Microsoft.Hosting.Lifetime|Hosting environment: Production
2024-05-07 06:47:26|ArchiSteamFarm-18|INFO|Microsoft.Hosting.Lifetime|Content root path: /asf
2024-05-07 06:47:26|ArchiSteamFarm-18|INFO|ASF|Start() IPC server ready!
2024-05-07 06:47:26|ArchiSteamFarm-18|INFO|JourneyOver|OnBotInitModules() Enable Free Packages : True
2024-05-07 06:47:26|ArchiSteamFarm-18|INFO|JourneyOver|OnBotInitModules() Free Packages Filters : [{"ImportStoreFilters":false,"Types":[],"Categories":[],"Tags":[],"IgnoredTypes":[],"IgnoredTags":[],"IgnoredCategories":[],"IgnoredContentDescriptors":[],"IgnoredAppIDs":[],"IgnoreFreeWeekends":false,"MinReviewScore":0,"Languages":[],"PlaytestMode":3,"RequireAllTags":false,"RequireAllCategories":false,"NoCostOnly":true,"Systems":[],"WishlistOnly":false}]
2024-05-07 06:47:26|ArchiSteamFarm-18|INFO|JourneyOver|Start() Starting...
2024-05-07 06:47:26|ArchiSteamFarm-18|INFO|JourneyOver|Connect() Connecting...
2024-05-07 06:47:27|ArchiSteamFarm-18|INFO|JourneyOver|OnConnected() Connected to Steam!
2024-05-07 06:47:27|ArchiSteamFarm-18|INFO|JourneyOver|OnConnected() Logging in...
2024-05-07 06:47:28|ArchiSteamFarm-18|INFO|JourneyOver|OnLoggedOn() Successfully logged on as <BLANKEDOUT>.
2024-05-07 06:47:28|ArchiSteamFarm-18|INFO|JourneyOver|Init() Success!
2024-05-07 06:47:31|ArchiSteamFarm-18|INFO|JourneyOver|IsAnythingToFarm() Checking first badge page...
2024-05-07 06:47:32|ArchiSteamFarm-18|INFO|JourneyOver|IsAnythingToFarm() Checking other badge pages...
2024-05-07 06:47:33|ArchiSteamFarm-18|INFO|JourneyOver|StartFarming() We don't have anything to farm on this account!
2024-05-07 06:47:52|ArchiSteamFarm-18|INFO|ASF|CollectGames() [FreeGames] found potentially 22 free games on reddit
2024-05-07 06:48:30|ArchiSteamFarm-18|INFO|JourneyOver|Refresh() Retrieving a total of 7 app access tokens...
2024-05-07 06:48:30|ArchiSteamFarm-18|INFO|JourneyOver|Refresh() Retrieving 7 app access tokens...
2024-05-07 06:48:30|ArchiSteamFarm-18|INFO|JourneyOver|Refresh() Finished retrieving 7 app access tokens.
2024-05-07 06:48:30|ArchiSteamFarm-18|INFO|JourneyOver|Refresh() Finished retrieving a total of 7 app access tokens.
2024-05-07 06:48:30|ArchiSteamFarm-18|INFO|JourneyOver|Refresh() Retrieving all depots for a total of 7 apps...
2024-05-07 06:48:31|ArchiSteamFarm-18|INFO|JourneyOver|Refresh() Retrieving 7 app infos...
2024-05-07 06:48:31|ArchiSteamFarm-18|INFO|JourneyOver|Refresh() Finished retrieving 7 app infos.
2024-05-07 06:48:32|ArchiSteamFarm-18|INFO|JourneyOver|Refresh() Successfully retrieved 1 out of 1 depot keys.
2024-05-07 06:48:32|ArchiSteamFarm-18|INFO|JourneyOver|Refresh() Finished retrieving all depot keys for a total of 7 apps.
2024-05-07 06:49:25|ArchiSteamFarm-18|INFO|ASF|ClaimItemTask() [JourneyOver] No reward to claim.
2024-05-07 07:00:25|ArchiSteamFarm-18|INFO|ASF|SubmitData() There is no new data to submit, everything is up-to-date.
2024-05-07 07:02:24|ArchiSteamFarm-18|FATAL|ASF|OnUnhandledException() System.MissingMethodException: Method not found: 'System.Text.RegularExpressions.Match System.Text.RegularExpressions.Regex.Match(System.String, System.String)'.
   at FreePackages.ASFInfo.DoUpdate()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)
   at FreePackages.ASFInfo.DoUpdate()
   at FreePackages.ASFInfo.<>c.<<-cctor>b__5_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_1(Object state)
   at System.Threading.QueueUserWorkItemCallbackDefaultContext.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
   at System.Threading.Thread.StartCallback()
2024-05-07 07:02:24|ArchiSteamFarm-18|ERROR|ASF|Exit() Exiting with 1 error code!
2024-05-07 07:02:24|ArchiSteamFarm-18|INFO|Microsoft.Hosting.Lifetime|Application is shutting down...
```

The application crashes/shuts down due to the error, and then restarts back up to repeat the same process again 15 minutes later.

**Comment by Citrinate** [comment]
I'm able to reproduce this and will have a fix for this later.  This is only an issue with ASF V6.0.3.0 and not the current stable release V6.0.2.6, so if possible downgrading will temporarily resolve the issue.

**Comment by Citrinate** [comment]
You can get the latest build with this fix here: [V1.4.5.1](https://github.com/Citrinate/FreePackages/actions/runs/8988788903/artifacts/1480975719)

**Comment by JourneyOver** [comment]
> You can get the latest build with this fix here: [V1.4.5.1](https://github.com/Citrinate/FreePackages/actions/runs/8988788903/artifacts/1480975719)

Sounds good, will test it out.
